### DISCUSSION TITLE
Amazon AWS problems

### DISCUSSION THREAD

**Discussion by SeppPenner** [dotnet/MQTTnet#1225] [Q&A]
Hello together, I would like to collect the AWS errors in one place as it seems like there are some issues that occur quite often.

Currently, there are the following resources to look up (Please look up before asking a question):
* Wiki, Connecting with AWS: https://github.com/chkr1011/MQTTnet/wiki/Client#connecting-with-amazon-aws
* Wiki, Troubleshooting AWS: https://github.com/chkr1011/MQTTnet/wiki/Client#troubleshooting
* Wiki, Using a client certificate: https://github.com/chkr1011/MQTTnet/wiki/Client#tls-using-a-client-certificate
* Snippet, Net Framework: https://gist.github.com/jkoplo/16f25875f7aca9503a7520fcda99005f
* Snippet, NetCore: https://gist.github.com/jkoplo/bd60cfe1a02c6e13b0a2d753289ae00f

Several related issues:
* https://github.com/chkr1011/MQTTnet/issues/109
* https://github.com/chkr1011/MQTTnet/issues/115
* https://github.com/chkr1011/MQTTnet/issues/141
* https://github.com/chkr1011/MQTTnet/issues/115
* https://github.com/chkr1011/MQTTnet/issues/202
* https://github.com/chkr1011/MQTTnet/issues/211
* https://github.com/chkr1011/MQTTnet/issues/267
* https://github.com/chkr1011/MQTTnet/issues/450
* https://github.com/chkr1011/MQTTnet/issues/452
* https://github.com/chkr1011/MQTTnet/issues/496
* https://github.com/chkr1011/MQTTnet/issues/612
* https://github.com/chkr1011/MQTTnet/issues/721
* https://github.com/chkr1011/MQTTnet/issues/736
* https://github.com/chkr1011/MQTTnet/issues/795
* https://github.com/chkr1011/MQTTnet/issues/894
* https://github.com/chkr1011/MQTTnet/issues/922
* https://github.com/chkr1011/MQTTnet/issues/927
* https://github.com/chkr1011/MQTTnet/issues/945
* https://github.com/chkr1011/MQTTnet/issues/970
* https://github.com/chkr1011/MQTTnet/issues/982
* https://github.com/chkr1011/MQTTnet/issues/1072
* https://github.com/chkr1011/MQTTnet/issues/1079
* https://github.com/chkr1011/MQTTnet/issues/1086
* https://github.com/chkr1011/MQTTnet/issues/1087
* https://github.com/chkr1011/MQTTnet/issues/1145
* https://github.com/chkr1011/MQTTnet/issues/1155

I will try to bundle all information here.

**Comment by snooksy** [comment]
since there are many questions about connecting to AWS then clearly people are still having problems based on the documentation. Wouldn't it be good to improve the documentation to show how to connect with CA and private cert?
